e2e/test/iothub/service/MessagingClientE2ETests.cs

```diff
@@ -122,6 +123,59 @@ public async Task MessagingClient_CanReopenClosedClient(IotHubTransportProtocol
             await sender.Messages.CloseAsync().ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        [DataRow(IotHubTransportProtocol.Tcp)]
+        [DataRow(IotHubTransportProtocol.WebSocket)]
+        public async Task MessagingClient_OpeningAlreadyOpenClient_DoesNotThrow(IotHubTransportProtocol protocol)
+        {
+            // arrange
+            var options = new IotHubServiceClientOptions
+            {
+                Protocol = protocol
+            };
+
+            await using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
+            using var serviceClient = new IotHubServiceClient(TestConfiguration.IotHub.ConnectionString, options);
+
+            // act
+            // Call OpenAsync on already open client
+            await serviceClient.Messages.OpenAsync().ConfigureAwait(false);
+            Func<Task> act = async() => await serviceClient.Messages.OpenAsync();
+
+            // assert
+            await act.Should().NotThrowAsync();
+        }
+
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        [DataRow(IotHubTransportProtocol.Tcp)]
+        [DataRow (IotHubTransportProtocol.WebSocket)]
+        public async Task MessagingClient_SendMessageOnClosedClient_ThrowsIotHubServiceException(IotHubTransportProtocol protocol)
```

Comment:
This seems to also be testing the client's internal behavior, independent of service's behavior. How about we refactor this into a unit test as well?

Reply:
Actually, it is interesting that we are expecting an IotHubServiceException and not an InvalidOperationException. Do you know what the underlying error message is? Since we've already closed the client it should be SDK generated.
I wonder if we should be throwing an InvalidOperationException instead.

Reply:
I agree `InvalidOperation` would be better. Changing it in `AmqpSesionHandler`